File: SEARCH/BINARY_SEARCH/BINARY_SEARCH.py
def binarySearch (arr, low, high, x):
    if high >= low:
        mid = (low + high)//2
        if arr[mid] == x:
            return mid
        elif arr[mid] > x:
            return binarySearch(arr, low, mid-1, x)
        else:
            return binarySearch(arr, mid+1, high, x)
    else:
        return -1
f=open('SEARCH\BINARY_SEARCH\inputs\input0.txt','r')
arr=list(map(int,f.readline().split()))
x=int(f.readline())
arr.sort()
result = binarySearch(arr, 0, len(arr)-1, x)
f1=open('SEARCH\BINARY_SEARCH\outputs\OUTPUT0.txt','w')
if result != -1:
    f1.write("Element is present at index "+ str(result))
else:
    f1.write("Element is not present in array")
f1.close()
''' 
Best Case Time Complexity: O(log(n)).
Space: O(1) in case of iterative implementation. In case of recursive implementation, O(Logn) recursion call stack space.

'''

File: SEARCH/BINARY_SEARCH/binary_search.cpp
#include <iostream>
using namespace std;
int binarySearch(int* arr, int low,int high,int k){
	if (high >= low){
        int mid = (low + high)/2;
        if (arr[mid] == k)
            return mid;
        else if (arr[mid] > k)
            return binarySearch(arr, low, mid-1, k);
        else
            return binarySearch(arr, mid+1, high, k);
	}
    else
        return -1;
}
int main(){
	int n,result;
	cout<<"Enter the sie of array: ";
	cin>>n;
	int* arr = new int[n];
	cout<<"Enter the array elements in sorted order: ";
	for(int i=0;i<n;i++){
		cin>>arr[i];
	}
	int k;
	cout<<"Enter the element to be searched: ";
	cin>>k;
	result= binarySearch(arr,0,n-1,k);
	if(result!= -1){
		cout<<"Element found at "<<result;
	}else{
		cout<<"ELement not found!";
	}
	return 0;
	
}
